storage/innobase/srv/srv0start.cc

```diff
@@ -2932,12 +2932,19 @@ struct metadata_applier {
   void operator()(dict_table_t *table) const {
     ut_ad(dict_sys->dynamic_metadata != nullptr);
     ib_uint64_t autoinc = table->autoinc;
+    ib_uint64_t autoinc_persisted = table->autoinc_persisted;
     dict_table_load_dynamic_metadata(table);
     /* For those tables which were not opened by
     ha_innobase::open() and not initialized by
     innobase_initialize_autoinc(), the next counter should be
-    advanced properly */
+    advanced properly.  */
+#if defined(XTRABACKUP)
+    if ((autoinc != table->autoinc ||
+         autoinc_persisted != table->autoinc_persisted) &&
```

Comment:
`autoinc_persisted != table->autoinc_persisted`
This condition will always be false as you are setting `autoinc_persisted=table->autoinc_persisted` on line 2935. Am I missing something?

Reply:
am not convinced with this change because this is the not the right way to determine autoinc or the way to load autoinc value